docs/the-new-architecture/pillars-turbomodule.md

```diff
@@ -3,16 +3,699 @@ id: pillars-turbomodules
 title: TurboModules
 ---
 
-This section contains a high-level introduction to TurboModules. It provides enough context to understand when a TurboModule is needed and how it roughly works.
+import Tabs from '@theme/Tabs'; import TabItem from '@theme/TabItem'; import constants from '@site/core/TabsConstants';
 
-This section must have a warning that it works only with the new architecture enabled. It points to the [migration section](../new-architecture-intro).
+If you've worked with React Native, you may be familiar with the concept of Native Modules, which allow JavaScript and platform-native code to communicate over the React Native "bridge", which handles cross-platform serialization via JSON.
 
-## How to create a Turbomodule
+TurboModules are the next iteration on Native Modules that provide a few extra [benefits](./why):
 
-This section is a step-by-step guide to create a TurboModule from scratch. The list of subsections is roughly:
+- Strongly typed interfaces that are consistent across platforms
+- The ability to create shared C++ code for use across platforms
+- Lazy loading of modules, allowing for faster app startup
+- The use of JSI, a JavaScript interface for native code, which allows for more efficient communication between native and JavaScript code than the bridge
 
-- JS spec (with all the supported features)
-- Configuration (package.json, cocoapods, gradle, …) and CodeGen
-- Native code (one section for iOS and one for Android)
-- Integration in an App (`yarn add` and how to connect the JS specs to the app itself)
-- Troubleshooting (common issues and how to solve them)
+This guide will show you how to create a basic TurboModule.
+
+:::caution
+TurboModules only work with the **New Architecture** enabled.
+To migrate to the **New Architecture**, follow the [Migration guide](../new-architecture-intro)
+:::
+
+## How to Create a TurboModule
+
+To create a TurboModule, we need to:
+
+1. Define a set of JavaScript specifications.
+2. Configure the module and inspect the code created by Codegen.
+3. Write the native code to finish implementing the module.
+
+## 1. Folder Setup
+
+In order to keep the module decoupled from the app, it's a good idea to define the module separately from the app, and then add it as a dependency to your app later.
```

Comment:
nit: we can add it that this will make it easier to open-source it as a library afterwards if needed

Reply:
addressed